File: app/src/main/java/com/privat/contacts/di/modules/DatabaseModule.java
package com.privat.contacts.di.modules;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Room;

import com.privat.contacts.data.cache.UsersDao;
import com.privat.contacts.data.cache.database.AppDatabase;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DatabaseModule {
    @NonNull
    @Singleton
    @Provides
    public AppDatabase provideDatabase(@NonNull Context context) {
        return Room.databaseBuilder(context, AppDatabase.class, "privat_contacts")
                .build();
    }

    @Singleton
    @Provides
    public UsersDao provideUsersDao(@NonNull AppDatabase database) {
        return database.provideUsersDao();
    }
}

File: app/src/main/java/com/privat/contacts/di/modules/DatabaseModule.kt
package com.privat.contacts.di.modules

import android.content.Context
import androidx.room.Room.databaseBuilder
import com.privat.contacts.data.cache.UsersDao
import com.privat.contacts.data.cache.database.AppDatabase
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DatabaseModule {
    @Singleton
    @Provides
    fun provideDatabase(context: Context): AppDatabase {
        return databaseBuilder(context, AppDatabase::class.java, "privat_contacts")
                .build()
    }

    @Singleton
    @Provides
    fun provideUsersDao(database: AppDatabase): UsersDao {
        return database.provideUsersDao()
    }
}
